src/Filesystem/FilesystemServiceProvider.php

```diff
@@ -60,5 +66,26 @@ public function register()
                 $container->make('flarum.filesystem.resolved_drivers')
             );
         });
+
+        $this->container->singleton(ImageManager::class, function (Container $container) {
+            /** @var Config $config */
+            $config = $this->container->make(Config::class);
+
+            $intervention = $config->offsetGet('intervention');
+            $driver = Arr::get($intervention, 'driver', self::INTERVENTION_DRIVERS['gd']);
+
+            // Check that the imagick library is actually available, else default back to gd.
+            if ($driver === self::INTERVENTION_DRIVERS['imagick'] && ! extension_loaded(self::INTERVENTION_DRIVERS['imagick'])) {
+                $driver = self::INTERVENTION_DRIVERS['gd'];
+            }
+
+            if (! Arr::has(self::INTERVENTION_DRIVERS, $driver)) {
+                throw new ValidationException(['intervention' => 'invalid driver']);
```

Comment:
this should be a runtime exception, since it's not user input, it's application configuration.